### PR TITLE
adding RAPIDS HPO notebook

### PR DESCRIPTION
**Pull request by zronaghi**
Adding an example notebook for GPU-accelerated HPO using RAPIDS, the training scripts include single and multi-GPU XGBoost and RandomForest models.

cc @JohnZed @drabastomek @manuelreyesgomez

### CODE REVIEW

File: code/models/rapids/train_rapids.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import cuml
+
+from cuml import RandomForestClassifier as cuRF
+from cuml.preprocessing.model_selection import train_test_split
+from cuml.metrics.accuracy import accuracy_score
+
+from rapids_csp_azure import RapidsCloudML, PerfTimer
+from azureml.core.run import Run
+
+run = Run.get_context()
```
Comment (lostmygithubaccount) [inline code/models/rapids/train_rapids.py:33]: no azureml-* code allowed in training scripts - see contributing guidelines - log via mlflow

File: notebooks/rapids/hpo-rapids.ipynb [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core.dataset import Dataset\n",
+    "airline_ds = Dataset.File.from_files('https://airlinedataset.blob.core.windows.net/airline-20m/*')\n",
```
Comment (lostmygithubaccount) [inline notebooks/rapids/hpo-rapids.ipynb:57]: who maintains this blob?

**Review by lostmygithubaccount — CHANGES_REQUESTED** [review]
general comments: 

- need to run black and black-nb - see contributing guidelines
- rename the notebook - see contributing guidelines for naming conventions 
- Estimators are being deprecated - switch to using ScriptRunConfig
- other minor issues

**Comment by lostmygithubaccount** [comment]
@zronaghi thanks for the PR! there are some changes needed, but I think this is close enough that I can make those changes and we can get this merged soon - I'll make a PR to your fork with some changes

File: notebooks/rapids/hpo-rapids.ipynb (hunk 2 above)
Comment (zronaghi) [reply notebooks/rapids/hpo-rapids.ipynb:57]: NVIDIA, unless there's another blob that's recommended for this repo?

**Comment by zronaghi** [comment]
Sounds great, thanks @lostmygithubaccount!

File: notebooks/rapids/hpo-rapids.ipynb (hunk 2 above)
Comment (lostmygithubaccount) [reply notebooks/rapids/hpo-rapids.ipynb:57]: @meyetman is this something we could host in the `azureopendata` container? I think we want to ensure we have control over all data used in samples

Comment (zronaghi) [reply notebooks/rapids/hpo-rapids.ipynb:57]: It's a public dataset, so totally fine with us

**Comment by lostmygithubaccount** [comment]
how would you configure this to use multiple GPUs?

**Comment by zronaghi** [comment]
Changing the `compute` parameter to multi-GPU the `train_rapids` script will use dask-cudf and multi-GPU RF XGBoost models

**Comment by lostmygithubaccount** [comment]
thanks for the contribution @zronaghi! this is awesome

i'll do some cleanup work today to get this added to the readme with a description (changing how that's automated)

**Comment by zronaghi** [comment]
Thanks for the quick review @lostmygithubaccount and updated changes for this repo

**Pull request merged**